File: galileo/src/main/java/com/josedlpozo/galileo/picker/ui/ScreenRecordRequestActivity.java
package com.josedlpozo.galileo.picker.ui;

import android.app.Activity;
import android.content.Intent;
import android.media.projection.MediaProjectionManager;
import android.os.Bundle;
import com.josedlpozo.galileo.picker.overlays.ColorPickerOverlay;
import com.josedlpozo.galileo.picker.utils.PreferenceUtils;

public class ScreenRecordRequestActivity extends Activity {

    @Override protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        MediaProjectionManager mpm = getSystemService(MediaProjectionManager.class);
        startActivityForResult(mpm.createScreenCaptureIntent(), 42);
    }

    @Override protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        if (resultCode == RESULT_OK) {
            DesignerTools.INSTANCE.setScreenRecordPermissionData(resultCode, data);
            Intent newIntent = new Intent(this, ColorPickerOverlay.class);
            this.startService(newIntent);
            PreferenceUtils.ColorPickerPreferences.setColorPickerActive(this, true);
        }
        finish();
    }
}

File: galileo/src/main/java/com/josedlpozo/galileo/picker/ui/ScreenRecordRequestActivity.kt
/*
 * Copyright (C) 2016 The CyanogenMod Project
 *
 * Modified Work: Copyright (c) 2018 fr4nk1
 *
 * Modified Work: Copyright (c) 2018 josedlpozo
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.josedlpozo.galileo.picker.ui

import android.annotation.TargetApi
import android.app.Activity
import android.content.Context
import android.content.Intent
import android.media.projection.MediaProjectionManager
import android.os.Build
import android.os.Bundle
import com.josedlpozo.galileo.picker.utils.PreferenceUtils

internal class ScreenRecordRequestActivity : Activity() {

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val mpm = getSystemService(Context.MEDIA_PROJECTION_SERVICE) as MediaProjectionManager
        startActivityForResult(mpm.createScreenCaptureIntent(), 42)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
        if (resultCode == RESULT_OK) {
            DesignerTools.setScreenRecordPermissionData(resultCode, data)
            PreferenceUtils.ColorPickerPreferences.setColorPickerEnabled(this, true)
        }
        finish()
    }
}
